src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,87 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.springboot.datajpa.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import mate.academy.springboot.datajpa.service.ProductService;
+import mate.academy.springboot.datajpa.service.mapper.request.CategoryRequestDtoMapper;
+import mate.academy.springboot.datajpa.service.mapper.request.ProductRequestDtoMapper;
+import mate.academy.springboot.datajpa.service.mapper.response.ProductResponseDtoMapper;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private ProductService productService;
+    private ProductResponseDtoMapper responseDtoMapper;
+    private ProductRequestDtoMapper requestDtoMapper;
+    private CategoryRequestDtoMapper categoryRequestDtoMapper;
+    private CategoryService categoryService;
+
+    public ProductController(ProductService productService,
+                             ProductResponseDtoMapper responseDtoMapper,
+                             ProductRequestDtoMapper requestDtoMapper,
+                             CategoryRequestDtoMapper categoryRequestDtoMapper,
+                             CategoryService categoryService) {
+        this.productService = productService;
+        this.responseDtoMapper = responseDtoMapper;
+        this.requestDtoMapper = requestDtoMapper;
+        this.categoryRequestDtoMapper = categoryRequestDtoMapper;
+        this.categoryService = categoryService;
+    }
+
+    @GetMapping()
+    public List<ProductResponseDto> getAllWhereCategoriesIn(
+            @RequestParam List<String> category) {
+        List<Category> categories = category.stream()
+                .map(categoryService::getCategoryByName)
+                .collect(Collectors.toList());
+        return productService.getAllProductsInCategories(categories)
+                .stream()
+                .map(responseDtoMapper::mapToDto)
+                .collect(Collectors.toList());
```

Comment:
Cool u done my thoughts ) I check this method too)
It's good for this task when we need to find only in categories.
But Specification more flexible, don't forget ).